pkg/apikey/apikey.go

```
@@ -61,123 +57,76 @@ func New(organizationID string) (*Key, error) {
-	bytes, err := hex.DecodeString(encodedPrivateKey)
-	if err != nil {
-		return nil, err
-	}
-
-	dValue := new(big.Int).SetBytes(bytes)
-
-	publicKey := new(ecdsa.PublicKey)
-	privateKey := ecdsa.PrivateKey{
-		PublicKey: *publicKey,
-		D:         dValue,
-	}
-
-	// Derive the public key
-	privateKey.PublicKey.Curve = elliptic.P256()
-	privateKey.PublicKey.X, privateKey.PublicKey.Y = privateKey.PublicKey.Curve.ScalarBaseMult(privateKey.D.Bytes())
-
-	apiKey, err := FromECDSAPrivateKey(&privateKey)
-	if err != nil {
-		return nil, err
-	}
+	// supply metadata
+	apiKey.Metadata.Organizations = append(apiKey.Metadata.Organizations, organizationID)
+	apiKey.Metadata.PublicKey = apiKey.PublicKey
+	apiKey.Metadata.Scheme = string(apiKey.scheme)
 
 	return apiKey, nil
 }
 
-// DecodeTurnkeyPublicKey takes a Turnkey-encoded public key and creates an ECDSA public key.
-func DecodeTurnkeyPublicKey(encodedPublicKey string) (*ecdsa.PublicKey, error) {
-	bytes, err := hex.DecodeString(encodedPublicKey)
-	if err != nil {
-		return nil, err
+// FromTurnkeyPrivateKey takes a Turnkey-encoded private key, derives a public key from it, and then returns the corresponding Turnkey API key.
+func FromTurnkeyPrivateKey(encodedPrivateKey string, scheme signatureScheme) (*Key, error) {
+	if scheme == SchemeED25519 {
+		return fromTurnkeyED25519Key(encodedPrivateKey)
 	}
 
-	if len(bytes) != 33 {
-		return nil, fmt.Errorf("expected a 33-bytes-long public key (compressed). Got %d bytes", len(bytes))
+	switch scheme {
+	case SchemeP256:
+		return fromTurnkeyECDSAKey(encodedPrivateKey, scheme)
+	case SchemeSECP256K1:
+		return fromTurnkeyECDSAKey(encodedPrivateKey, scheme)
+	case SchemeED25519:
+		return fromTurnkeyED25519Key(encodedPrivateKey)
+	default:
 	}
 
-	x, y := elliptic.UnmarshalCompressed(elliptic.P256(), bytes)
-
-	return &ecdsa.PublicKey{
-		Curve: elliptic.P256(),
-		X:     x,
-		Y:     y,
-	}, nil
+	return nil, errors.New("unsupported signature scheme")
```

Since you have a `default`, can we return this error there?